src/_gettsim/transfers/wohngeld.py

```diff
@@ -13,37 +21,55 @@
 }
 
 
-def wohngeld_m_hh(
-    wohngeld_nach_vermög_check_m_hh: float,
-    wohngeld_vorrang_hh: bool,
-    wohngeld_kinderzuschl_vorrang_hh: bool,
+def wohngeld_m_wthh(
+    _wohngeld_nach_vermög_check_m_wthh: float,
     erwachsene_alle_rentner_hh: bool,
+    wohngeld_kinderzuschl_vorrang_wthh: bool,
+    wohngeld_vorrang_wthh: bool,
 ) -> float:
-    """Calculate final housing benefit on household level.
+    """Calculate housing benefit.
+
+    Wohngeld has priority over ALG2 if the recipients can cover their needs according to
+    SGB II when receiving Wohngeld. The priority check follows the following logic:
+
+    1. Calculate Wohngeld on the Bedarfsgemeinschaft level.
+    2. Check whether the Bedarfsgemeinschaft can cover its own needs (Regelbedarf) with
+       Wohngeld. If not, the Bedarfsgemeinschaft is eligible for ALG2.
+    3. Compute Wohngeld again for all individuals in the household that can cover their
+       own needs with Wohngeld. This is the final Wohngeld amount that is paid out to
+       the wohngeldrechtlicher Teilhaushalt.
+
+    Note: Because Wohngeld is nonlinear in the number of people in the
+    wohngeldrechtlicher Teilhaushalt, there may be some individuals that pass the
+    priority check, but cannot cover their needs with the Wohngeld calculated in point
+    3. In this sense, this implementation is an approximation of the actual Wohngeld.
 
     Parameters
     ----------
-    wohngeld_nach_vermög_check_m_hh
-        See :func:`wohngeld_nach_vermög_check_m_hh`.
-    wohngeld_vorrang_hh
-        See :func:`wohngeld_vorrang_hh`.
-    wohngeld_kinderzuschl_vorrang_hh
-        See :func:`wohngeld_kinderzuschl_vorrang_hh`.
+    _wohngeld_nach_vermög_check_m_wthh
+        See :func:`_wohngeld_nach_vermög_check_m_wthh`.
     erwachsene_alle_rentner_hh
-        See :func:`erwachsene_alle_rentner_hh`.
+        See :func:`erwachsene_alle_rentner_hh <erwachsene_alle_rentner_hh>`.
+    wohngeld_kinderzuschl_vorrang_wthh
+        See :func:`wohngeld_kinderzuschl_vorrang_wthh`.
+    wohngeld_vorrang_wthh
+        See :func:`wohngeld_vorrang_wthh`.
 
     Returns
     -------
 
     """
-    if (
-        (not wohngeld_vorrang_hh)
-        and (not wohngeld_kinderzuschl_vorrang_hh)
-        or erwachsene_alle_rentner_hh
+    # TODO (@MImmesberger): Document approximation.
+    # TODO (@MImmesberger): Remove erwachsene_alle_rentner_hh condition once households
```

Comment:
This confuses me. Isn't the problem rather that we don't have an SGB XII / WoG priority check just yet?

Reply:
I'm not sure tbh, just tried to avoid touching that part because I don't know what the actual problem 2 years ago was when that check was introduced. It could be that the problems are resolved now that Wohngeld is calculated on a more granular level than `hh` and we can, thus, compare it to the SGB XII transfer.

Reply:
Yeah, sounds like a good place to put it.

Reply:
I think it is still relevant since we haven't implemented any interaction between Grundsicherung im Alter and ALG2/Wohngeld.

In particular, I have no idea what happens if one partner is retired and eligible for Grundsicherung im Alter and the other one is not retired and eligible for ALG2 or eligible for Wohngeld.

Reply:
Yes, this is what I had been trying to say -- the comment just is very opaque in this respect.

Reply:
Yes, I agree.

Reply:
Adjusted the todo note accordingly.